server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -416,6 +444,12 @@ protected Status replicateLogs(ClientContext peerContext, final HostAndPort peer
     Status lastStatus = status, currentStatus = status;
     final AtomicReference<Exception> exceptionRef = new AtomicReference<>();
     while (true) {
+      // Check if the thread has been interrupted prior to replicating data
+      if (Thread.interrupted()) {
```

Comment:
I can't be sure if its needed.  Its a question of do all of the functions called by this code handle interrupts properly? I have no idea what the answer to that question is.